site/gulpfile.js

```diff
@@ -80,9 +81,10 @@ gulp.task('swag-img:update-data', () => {
         s.image = `/assets/swag-img/${name}`;
         return s;
     });
-    const payload = JSON.stringify(newSwagList);
-    fs.writeFileSync('dist/assets/data.json', payload);
     SWAG_LIST = newSwagList;
+    const stream = source('data.json');
```

Comment:
Is there a reason to use streams here in place of fs? Since Gulp supports callbacks (and promises as well, but using callbacks here for simplicity)

```js
const {writeFile} = require('fs');

// ...
gulp.task(/*...*/, cb =>
/*...*/
return writeFile('./dist/assets/data.json', JSON.stringify(newSwagList), cb);
```

Reply:
I used streams to be sure that the folder is being created by gulp. But I can revert the commit and use asynchronous functions instead, sure.

Reply:
I agree that adding a dependency for this unique thing may not be necessary

Reply:
Do you think this is better ?
```js
return mkdir('dist/assets', (err) => {
	// if (err) throw err;
	writeFile('dist/assets/data.json', JSON.stringify(newSwagList), cb)
});
```